bin/btc-server/src/util.rs

```
@@ -466,10 +507,27 @@ pub(crate) fn validate_outputs(psbt: &Psbt, db: &database::Db) -> Result<(), Val
         };
     }
 
-    // check outputs are not in finalized pegouts list
+    // check outputs are not in finalized pegouts list and not too old
     let finalized_pegouts_id =
         db.get_finalized_pegout_ids()?.into_iter().map(|id| id.id).collect::<Vec<_>>();
+
     for id in &psbt_pegout_ids {
+        if botanix_eth_client.is_connected().await {
```

Afaik this will just skip the `is_block_age_acceptable` check entirely if the client is not connected? It should return an error instead, or similar.

---

@lamafab There is an issue here of a cyclic dependency - you need your btc-server to run first before the core node can start. If I remove the connected part then we would panic / fail here and the btc-server wont start leading to the reth node not being to start. With this "Semi-conservative" check we let the btc-server start first - in this process obviously the above returning not connected and no checks being performed. Once it is however up, then the reth node is able to start and a few secs down the line the connection above is already up and the checks gets triggered. With the current overall architecture there is no other way to get around this issue. I think this is an elegant solution which works pretty  well considering what we have. cc @scottmillner